component/azstorage/datalake.go

```diff
@@ -118,6 +118,7 @@ func (dl *Datalake) UpdateConfig(cfg AzStorageConfig) error {
 	dl.Config.maxConcurrency = cfg.maxConcurrency
 	dl.Config.defaultTier = cfg.defaultTier
 	dl.Config.ignoreAccessModifiers = cfg.ignoreAccessModifiers
```

Comment:
ignoreAccessModifiers is also doing someting similar, lets consolidate both the flags with "preserve-permissions" name. By default flag will be set to 'default'. In block_blob.go it shall ignore both chmod and chown operations while in case of datalake chown shall be ignored. If flag is set explicitly to false then datalake shall ignore chmod as well, rest remains same as default. If flag is set explicitly to true then we shall return back ENOSUP from block_blob and chown in datalake.

Reply:
preserve-acl flag is to only preserve ACLs while overwriting a file. Also, ACLs and permissions are different things so tommorow we may need a differentiation between these two for when to preserve what